pkg/controller/operators/olm/operator.go

```diff
@@ -319,6 +322,21 @@ func newOperatorWithConfig(ctx context.Context, config *operatorConfig) (*Operat
 		if err := op.RegisterQueueInformer(serviceAccountQueueInformer); err != nil {
 			return nil, err
 		}
+
+		objGCQueue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), fmt.Sprintf("%s/obj-gc", namespace))
+		objGCQueueInformer, err := queueinformer.NewQueue(
+			ctx,
+			queueinformer.WithLogger(op.logger),
+			queueinformer.WithQueue(objGCQueue),
+			queueinformer.WithSyncer(queueinformer.LegacySyncHandler(op.syncGCObject).ToSyncer()),
```

Comment:
Is using the legacy adapter more convenient than implementing a new `Syncer`? If that's the case then we may want to remove that interface going forward.

Reply:
Since I'm familiar with the "old" way, I was going to implement that first.